examples/preprocessing/plot_csd.py

```diff
@@ -0,0 +1,139 @@
+Analyze Data Using Current Source Density (CSD)
+=================================================================
+
+This script shows an example of how to use CSD. CSD
+takes the spatial Laplacian of the sensor signal (derivative in both
+x and y). It does what a planar gradiometer does in MEG. Spatial derivative
+reduces point spread. CSD transformed data have a sharper or more distinct topography,
+reducing the negative impact of volume conduction.
+"""
+# Authors: Alex Rockhill <aprockhill206@gmail.com>
+#
+# https://gist.github.com/sherdim/28b069c8ebca17073f24322e8e721e1f
+
+# License: BSD (3-clause)
+import matplotlib.pyplot as plt
+
+import numpy as np
+import mne
+from mne.datasets import sample
+
+from mne.preprocessing import compute_current_source_density
+
+print(__doc__)
+
+###############################################################################
+# Load sample subject data
+data_path = sample.data_path()
+
+raw = mne.io.read_raw_fif(data_path + '/MEG/sample/sample_audvis_raw.fif',
+                          preload=True)
+raw = raw.pick_types(meg=False, eeg=True, eog=True, ecg=True, stim=True,
+                     exclude=raw.info['bads'])
+event_id = {'auditory/left': 1, 'auditory/right': 2, 'visual/left': 3,
+            'visual/right': 4, 'smiley': 5, 'button': 32}
+events = mne.find_events(raw)
+epochs = mne.Epochs(raw, events, event_id=event_id, tmin=-0.2, tmax=.5,
+                    preload=True)
+
+###############################################################################
+# Plot sensor relations
+
+evo = epochs.average()
+evo.plot_sensors(show_names=True)
+
+###############################################################################
+# Laplacian depends on spacial relations, let's look at EEG 060
+
+evo2 = evo.copy().drop_channels(['EEG 060'])
+evo2.plot_sensors(show_names=['EEG 032', 'EEG 014', 'EEG 030', 'EEG 034',
+                              'EEG 050'])
+
+###############################################################################
+# Here are the time courses
+
+fig = evo.plot(picks=np.array([32, 14, 30, 34, 50]) - 1,
+               spatial_colors=True, time_unit='s', show=False)
+fig.findobj(plt.Line2D)[0].set_linewidth(3)
+plt.show()
+
+###############################################################################
+# Here is some literature:
+# Perrin et al. (1989, 1990)
+#
+# \MATLAB\eeglab14_1_1b\functions\popfunc\eeg_laplac.m
+#    Juan Sebastian Gonzalez, DFI / Universidad de Chile
+#
+# MATLAB\eeglab14_1_1b\plugins\PrepPipeline0.55.3\utilities\private\spherical_interpolate.m
+#    Copyright 2009-     by Jason D.R. Farquhar (jdrf@zepler.org)
+#
+# Wang K, Begleiter, 1999
+#
+# MATLAB\eeglab14_1_1b\plugins\erplab6.1.3\functions\csd\current_source_density.m
+#    Copyright (C) 2003 by Jürgen Kayser (Email: kayserj@pi.cpmc.columbia.edu)
+#    (published in appendix of Kayser J, Tenke CE,
+#     Clin Neurophysiol 2006;117(2):348-368)
```

Comment:
None of this should be in the example. But you should have a `References` section at the bottom with refs